exciting_environments/core_env.py

```diff
@@ -149,109 +143,101 @@ def _test_rew_func(self, func):
+                    len(list(vars(self.env_properties.action_constraints).values()))))),
+                self.env_properties.action_constraints
+            )
         except:
             raise Exception(
-                "Reward function should be using obs matrix as only parameter")
+                "Reward function should be using observation matrix, action and self.env_properties.action_constraints as parameters")
         try:
             if out.shape != (self.batch_size, 1):
                 raise Exception(
-                    "Reward function should be returning vector in shape (batch_size,1)")
+                    "Reward function should be returning vector in shape (1,)")
         except:
             raise Exception(
-                "Reward function should be returning vector in shape (batch_size,1)")
+                "Reward function should be returning vector in shape (1,)")
         return True
 
     @partial(jax.jit, static_argnums=0)
-    def step(self, action_norm, states):
-        """Addtional function in step execution to enable JAX jit.
+    def step(self, states, action, env_properties):
+        """Computes one simulation step for one batch.
 
         Args:
-            states(ndarray(float)): State Matrix (shape=(batch_size,states)).
-            action_norm(ndarray(float)): Action Matrix (shape=(batch_size,actions)).
-
+            states: The states from which to calculate states for the next step.
+            action: The action to apply to the environment.
+            env_properties: Contains action/state constraints and static parameter.
 
         Returns:
             Multiple Outputs:
 
-            observation(ndarray(float)): Observation/State Matrix (shape=(batch_size,states)).
-
-            reward(ndarray(float)): Amount of reward received for the last step (shape=(batch_size,1)).
+            observation: The gathered observation.
 
-            terminated(bool): Flag, indicating if Agent has reached the terminal state.
+            reward: Amount of reward received for the last step.
 
-            truncated(ndarray(bool)): Flag, indicating if state has gone out of bounds (shape=(batch_size,states)).
+            terminated: Flag, indicating if Agent has reached the terminal state.
 
-            {}: An empty dictionary for consistency with the OpenAi Gym interface.
+            truncated: Flag, indicating if state has gone out of bounds.
 
+            states: New states for the next step.
         """
+
         # ode step
-        states = jax.vmap(self._ode_exp_euler_step)(states, action_norm, self.env_state_normalizer,
-                                                    self.action_normalizer, self.static_params)
+        states = self._ode_solver_step(
+            states, action, env_properties.static_params)
 
         # observation
-        obs = jax.vmap(self.generate_observation)(states)
+        obs = self.generate_observation(
+            states, env_properties.physical_constraints)
```

Comment:
This is something that I have seen a few times already, so this is not specific to this function call here.

Following the [python google style guide's section on line breaks](https://google.github.io/styleguide/pyguide.html#32-line-length) my preferred way to do line breaks is like this:

```suggestion
        obs = self.generate_observation(
            states, env_properties.physical_constraints
        )
```

```
output = func(a, b, c, foo, bar, very_long_input_so_that_a_new_line_is_necessary,
              states, foo2, bar2)
```

is also absolutely valid. But this way of formatting does not work so well in this specific case.

What you are doing a lot is a bit of a combination of the two, which I do not really like as it is less
clear which brackets belong together IMO.